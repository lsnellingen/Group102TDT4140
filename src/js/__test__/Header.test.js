// Libs
import React from 'react';
import { Link } from 'react-router';
import { LoginLink, LogoutLink, NotAuthenticated, Authenticated } from 'react-stormpath';

import renderer from 'react-test-renderer';
import Header from '../pages/Header';
import {shallow} from 'enzyme';

 describe('Header', () => {
  var ReactTestUtils = require('react-addons-test-utils') // ES5 with npm

    test('renders without crashing', () => {


      const tree = shallow(
        <Header/>);


      expect(tree).toMatchSnapshot();




    });
});
