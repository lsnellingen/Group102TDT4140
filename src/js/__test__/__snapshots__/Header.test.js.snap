// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <nav
    className="navbar topbar navbar-default navbar-static-top removeMargin"
>
    <div
        className="container-fluid col-xs-8 col-xs-offset-2 headerAlign"
    >
        <div
            className="navbar-header"
        >
            <button
                className="navbar-toggle collapsed"
                data-target="#navbar-collapse"
                data-toggle="collapse"
                type="button"
            >
                <span
                    className="sr-only"
                >
                    Toggle Navigation
                </span>
                <span
                    className="icon-bar"
                />
                <span
                    className="icon-bar"
                />
                <span
                    className="icon-bar"
                />
            </button>
        </div>
        <div
            className="collapse navbar-collapse"
            id="navbar-collapse"
        >
            <ul
                className="nav navbar-nav"
            >
                <Authenticated>
                    <li>
                        <Link
                            activeClassName="active"
                            onlyActiveOnIndex={false}
                            style={Object {}}
                            to="/profile"
                        >
                            Profile
                        </Link>
                    </li>
                </Authenticated>
            </ul>
            <ul
                className="nav navbar-nav navbar-right"
            >
                <NotAuthenticated>
                    <li>
                        <LoginLink
                            activeClassName="active"
                        />
                    </li>
                </NotAuthenticated>
                <Authenticated>
                    <li>
                        <LogoutLink />
                    </li>
                </Authenticated>
                <NotAuthenticated>
                    <li>
                        <Link
                            activeClassName="active"
                            onlyActiveOnIndex={false}
                            style={Object {}}
                            to="/register"
                        >
                            Register
                        </Link>
                    </li>
                </NotAuthenticated>
            </ul>
        </div>
    </div>
</nav>,
  "nodes": Array [
    <nav
      className="navbar topbar navbar-default navbar-static-top removeMargin"
>
      <div
            className="container-fluid col-xs-8 col-xs-offset-2 headerAlign"
      >
            <div
                  className="navbar-header"
            >
                  <button
                        className="navbar-toggle collapsed"
                        data-target="#navbar-collapse"
                        data-toggle="collapse"
                        type="button"
                  >
                        <span
                              className="sr-only"
                        >
                              Toggle Navigation
                        </span>
                        <span
                              className="icon-bar"
                        />
                        <span
                              className="icon-bar"
                        />
                        <span
                              className="icon-bar"
                        />
                  </button>
            </div>
            <div
                  className="collapse navbar-collapse"
                  id="navbar-collapse"
            >
                  <ul
                        className="nav navbar-nav"
                  >
                        <Authenticated>
                              <li>
                                    <Link
                                          activeClassName="active"
                                          onlyActiveOnIndex={false}
                                          style={Object {}}
                                          to="/profile"
                                    >
                                          Profile
                                    </Link>
                              </li>
                        </Authenticated>
                  </ul>
                  <ul
                        className="nav navbar-nav navbar-right"
                  >
                        <NotAuthenticated>
                              <li>
                                    <LoginLink
                                          activeClassName="active"
                                    />
                              </li>
                        </NotAuthenticated>
                        <Authenticated>
                              <li>
                                    <LogoutLink />
                              </li>
                        </Authenticated>
                        <NotAuthenticated>
                              <li>
                                    <Link
                                          activeClassName="active"
                                          onlyActiveOnIndex={false}
                                          style={Object {}}
                                          to="/register"
                                    >
                                          Register
                                    </Link>
                              </li>
                        </NotAuthenticated>
                  </ul>
            </div>
      </div>
</nav>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Header />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Header {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <nav
          className="navbar topbar navbar-default navbar-static-top removeMargin"
>
          <div
                    className="container-fluid col-xs-8 col-xs-offset-2 headerAlign"
          >
                    <div
                              className="navbar-header"
                    >
                              <button
                                        className="navbar-toggle collapsed"
                                        data-target="#navbar-collapse"
                                        data-toggle="collapse"
                                        type="button"
                              >
                                        <span
                                                  className="sr-only"
                                        >
                                                  Toggle Navigation
                                        </span>
                                        <span
                                                  className="icon-bar"
                                        />
                                        <span
                                                  className="icon-bar"
                                        />
                                        <span
                                                  className="icon-bar"
                                        />
                              </button>
                    </div>
                    <div
                              className="collapse navbar-collapse"
                              id="navbar-collapse"
                    >
                              <ul
                                        className="nav navbar-nav"
                              >
                                        <Authenticated>
                                                  <li>
                                                            <Link
                                                                      activeClassName="active"
                                                                      onlyActiveOnIndex={false}
                                                                      style={Object {}}
                                                                      to="/profile"
                                                            >
                                                                      Profile
                                                            </Link>
                                                  </li>
                                        </Authenticated>
                              </ul>
                              <ul
                                        className="nav navbar-nav navbar-right"
                              >
                                        <NotAuthenticated>
                                                  <li>
                                                            <LoginLink
                                                                      activeClassName="active"
                                                            />
                                                  </li>
                                        </NotAuthenticated>
                                        <Authenticated>
                                                  <li>
                                                            <LogoutLink />
                                                  </li>
                                        </Authenticated>
                                        <NotAuthenticated>
                                                  <li>
                                                            <Link
                                                                      activeClassName="active"
                                                                      onlyActiveOnIndex={false}
                                                                      style={Object {}}
                                                                      to="/register"
                                                            >
                                                                      Register
                                                            </Link>
                                                  </li>
                                        </NotAuthenticated>
                              </ul>
                    </div>
          </div>
</nav>,
        "_debugID": 2,
        "_renderedOutput": <nav
          className="navbar topbar navbar-default navbar-static-top removeMargin"
>
          <div
                    className="container-fluid col-xs-8 col-xs-offset-2 headerAlign"
          >
                    <div
                              className="navbar-header"
                    >
                              <button
                                        className="navbar-toggle collapsed"
                                        data-target="#navbar-collapse"
                                        data-toggle="collapse"
                                        type="button"
                              >
                                        <span
                                                  className="sr-only"
                                        >
                                                  Toggle Navigation
                                        </span>
                                        <span
                                                  className="icon-bar"
                                        />
                                        <span
                                                  className="icon-bar"
                                        />
                                        <span
                                                  className="icon-bar"
                                        />
                              </button>
                    </div>
                    <div
                              className="collapse navbar-collapse"
                              id="navbar-collapse"
                    >
                              <ul
                                        className="nav navbar-nav"
                              >
                                        <Authenticated>
                                                  <li>
                                                            <Link
                                                                      activeClassName="active"
                                                                      onlyActiveOnIndex={false}
                                                                      style={Object {}}
                                                                      to="/profile"
                                                            >
                                                                      Profile
                                                            </Link>
                                                  </li>
                                        </Authenticated>
                              </ul>
                              <ul
                                        className="nav navbar-nav navbar-right"
                              >
                                        <NotAuthenticated>
                                                  <li>
                                                            <LoginLink
                                                                      activeClassName="active"
                                                            />
                                                  </li>
                                        </NotAuthenticated>
                                        <Authenticated>
                                                  <li>
                                                            <LogoutLink />
                                                  </li>
                                        </Authenticated>
                                        <NotAuthenticated>
                                                  <li>
                                                            <Link
                                                                      activeClassName="active"
                                                                      onlyActiveOnIndex={false}
                                                                      style={Object {}}
                                                                      to="/register"
                                                            >
                                                                      Register
                                                            </Link>
                                                  </li>
                                        </NotAuthenticated>
                              </ul>
                    </div>
          </div>
</nav>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Header />,
}
`;
